keps/sig-network/3766-referencegrant/README.md

```diff
@@ -0,0 +1,697 @@
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+[ReferenceGrant](https://gateway-api.sigs.k8s.io/api-types/referencegrant/) was
+developed by the [Gateway API subproject](https://gateway-api.sigs.k8s.io/) to
+enable certain object references to cross namespaces. More recently, it has also
+been [used by
+sig-storage](https://kubernetes.io/blog/2023/01/02/cross-namespace-data-sources-alpha/)
+to enable cross-namespace data sources.
+
+This KEP proposes moving ReferenceGrant from its current
+`gateway.networking.k8s.io` API group into the `authorization.k8s.io` API
+group.
+
+## Motivation
+
+Any project that wants to enable cross-namespace references currently has to choose
+between introducing a dependency on Gateway API's ReferenceGrant or creating a
+new API that would be partially redundant (leading to confusion for users).
+	
+Recent interest between SIGs has made it clear that ReferenceGrant is wanted for use
+cases other than Gateway API. We would like to move ReferenceGrant to a neutral home
+(ideally, under sig-auth) in order to make it the canonical API for managing references
+across namespaces.
+
+### Goals
+
+* Move ReferenceGrant to sig-auth API Group
+* Clearly define how ReferenceGrant should be used, including both current use
+  cases and guidance for future use cases
+* Implement a library to ensure that ReferenceGrant is implemented consistently
+  by all controllers
+
+### Non-Goals
+
+* Add, remove, or change fields in the ReferenceGrant API
+* Develop an authorizer that will automatically implement ReferenceGrant for
+  all use cases
+
+## Proposal
+
+Move the existing ReferenceGrant resource into a new
+`authorization.k8s.io` API group, defined within the Kubernetes code base
+as part of the 1.27 release.
+
+We will take this opportunity to clarify and update the API after SIG-Auth
+feedback. This resource will start with v1alpha1 as the API version.
```

Comment:
Will we also provide an official CRD for people who want to use the new API on older clusters?

Reply:
I like the sound of this, but I don't know anything about how it would work. 😄

Reply:
I think the idea that this resource will also include an authorization layer that insures that the creator of the resource also has read or write access to the resource they're granting access to would make this pretty tough. Instead I think we'd just recommend using the CRD already provided by Gateway API, which unfortunately will be slightly different than what SIG Auth accepts here.